eslint.config.js

```diff
@@ -0,0 +1,75 @@
+// @ts-check
+
+import eslint from '@eslint/js';
+import tseslint from 'typescript-eslint';
+import {join, dirname} from 'node:path';
+import {fileURLToPath} from 'node:url';
+
+// TODO: Replace with import.meta.dirname after Node.js v18 EOL, May 2025.
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+/** @type {Partial<Record<string, any>>} */
+const PRODUCTION_RULES = {
+  // TODO: These rules should be re-enabled for production code.
+  '@typescript-eslint/no-explicit-any': 'off',
+  '@typescript-eslint/no-unsafe-argument': 'off',
+  '@typescript-eslint/no-unsafe-assignment': 'off',
+  '@typescript-eslint/no-unsafe-call': 'off',
+  '@typescript-eslint/no-unsafe-member-access': 'off',
+  '@typescript-eslint/no-unsafe-return': 'off',
+  '@typescript-eslint/ban-ts-comment': 'off',
+};
+
+/** @type {Partial<Record<string, any>>} */
+const DEVELOPMENT_RULES = {
+  '@typescript-eslint/no-explicit-any': 'off',
+  '@typescript-eslint/no-unsafe-argument': 'off',
+  '@typescript-eslint/no-unsafe-assignment': 'off',
+  '@typescript-eslint/no-unsafe-call': 'off',
+  '@typescript-eslint/no-unsafe-member-access': 'off',
+  '@typescript-eslint/no-unsafe-return': 'off',
+
+  // TODO: These rules should be re-enabled for all code.
+  '@typescript-eslint/ban-ts-comment': 'off',
+};
```

Comment:
Thinking out loudly, I think we should start thinking about creating some kind of `@carto/eslint-config`, because I think codebases should be aligned within these rules, and maintaining only an ESLint Config seems easier, safer, and better, for all of us, WDYT?

Said so, thank you very much for caring about DEV/PROD rules, I think this aligns pretty well with the mess in `cloud-native` where we disable the same ESLint rules on every test, great job! ❤️